### ISSUE TITLE
Intensity tolerance in Peak finder module

### ISSUE THREAD

**Issue by NthlLcrmp** [mzmine/mzmine3#1279]
Dear all,

I don't understand the "Intensity tolerance" parameter of the Peak finder module (gap-filling). 

The MZmine publication explains about the gap-filling module: 
"This method first searches for a local intensity maximum within a selected chromatographic region corresponding to expected location of a missed peak, which is used as an estimate for peak height. The peak area estimate is then calculated by moving from the maximum to both directions along the extracted ion chromatogram as long as the peak curve is monotonously decreasing within the pre-specified tolerance limits.".
Despite these explanations I don't understand what it means. Could someone explain it to me?

I observed that the higher the percentage, the less missing values but the integrated peaks look more like noise than real peaks. 
What is the "risk" of setting this parameter to 100% to have as few missing values as possible (beside integrated noise of course)?

Thank you for your help

Regards,
Nathalie

**Comment by robinschmid** [comment]
hey Nathalie,

the percentage that you provide is a tolerance for the intensity. Lets say we go down from the top of the peak on one side and we reach a local minimum and the intensity goes up again after this - if the intensity reaches >+20% of the lowest intensity we end the feature. This is done on both sides. 

The implementation details are in the code. But that's roughly how it works.

Cheers
Robin

**Comment by robinschmid** [comment]
As for your question, 
If you have multiple peaks that are not baseline separated, you will end up using the whole area. This will only skew the statistical analysis. But missing a feature might as well.

**Comment by NthlLcrmp** [comment]
Thank you Robin for these explanations.

**Comment by keunbae318** [comment]
Hello Robin,

additionally, I also have a question that after gap-filling some of features mislocated although the corrected feature is visually closed to the group of aligned features (two candidate peaks are very similar and close). I cannot restrict retention time tolerance more than certain as broad peak and large peaks require a fairly enough time. 

**I was curious of what is the priority to assign gap-filling peaks, is there additional parameter need to be setting up?** (I guess I do all of them though; intensity, mz, retention tolerance, and minimum scans). I assume it should have selected peak more close to the aligned peak group based on the retention time if there are two similar one coexisting. 

Cheers,
Keunbae

**Comment by robinschmid** [comment]
Can you share some screenshots that describe the behavior and how you would expect it?

**Comment by jessdeng** [comment]
Hi @robinschmid, related to intensity tolerance, I'm wondering if there's any way for every single peak to be selected in an export? Although I have selected the noise level to be 0 under "mass detection" for the "factor of the lowest signal", when I copy my data to either the clipboard to copy into excel or export to excel it seems that it does not detect every mass. I've attached some screenshots. 

Thanks in advance! 
<img width="917" alt="Screenshot 2023-04-14 at 7 18 11 PM" src="https://user-images.githubusercontent.com/101413266/232169648-acbe0153-a034-403e-8be7-7f26d3a145f9.png">
<img width="116" alt="Screenshot 2023-04-14 at 7 19 59 PM" src="https://user-images.githubusercontent.com/101413266/232169654-d51b332a-f622-43aa-97f6-881c304e0688.png">
